src/track/serato/tags.cpp

```diff
@@ -145,71 +147,106 @@ QList<CueInfo> SeratoTags::getCues(const QString& filePath) const {
     // from "Serato Markers_". This is what Serato does too (i.e. if
     // "Serato Markers_" and "Serato Markers2" contradict each other,
     // Serato will use the values from "Serato Markers_").
+    //
+    //
+    // In Serato, loops and hotcues are kept separate (i. e. you can
+    // have a loop and a hotcue with the same number. In Mixxx, loops
+    // and hotcues share indices. Hence, we import them with and offset
+    // of 8 (the maximum number of hotcues in Serato).
+    //
 
     double timingOffsetMillis = SeratoTags::findTimingOffsetMillis(filePath);
 
     QMap<int, CueInfo> cueMap;
     for (const CueInfo& cueInfo : m_seratoMarkers2.getCues(timingOffsetMillis)) {
-        int index = 0;
+        VERIFY_OR_DEBUG_ASSERT(cueInfo.getHotCueNumber()) {
+            qWarning() << "SeratoTags::getCues: Cue without number found!";
+            continue;
+        }
+
+        int index = *cueInfo.getHotCueNumber();
+        VERIFY_OR_DEBUG_ASSERT(index >= 0) {
+            qWarning() << "SeratoTags::getCues: Cue with number < 0 found!";
+        }
+
         switch (cueInfo.getType()) {
-        case CueType::HotCue:
-            DEBUG_ASSERT(cueInfo.getHotCueNumber());
-            index = *cueInfo.getHotCueNumber();
-            DEBUG_ASSERT(index >= 0);
+        case CueType::HotCue: {
+            if (index >= kLoopIndexOffset) {
+                qWarning()
+                        << "SeratoTags::getCues: Non-loop Cue with number >="
+                        << kLoopIndexOffset << "found!";
+                continue;
+            }
             break;
-        case CueType::Loop:
+        }
+        case CueType::Loop: {
+            index += kLoopIndexOffset;
             break;
+        }
         default:
-            DEBUG_ASSERT(false);
-            continue;
+            break;
         }
 
+        CueInfo newCueInfo(cueInfo);
         RgbColor::optional_t color = cueInfo.getColor();
         if (color) {
             // TODO: Make this conversion configurable
-            CueInfo newCueInfo = cueInfo;
             newCueInfo.setColor(storedToDisplayedSeratoDJProCueColor(*color));
-            cueMap.insert(index, newCueInfo);
-        } else {
-            cueMap.insert(index, cueInfo);
         }
+        newCueInfo.setHotCueNumber(index);
+        cueMap.insert(index, newCueInfo);
     };
 
-    // TODO: If a hotcue is set in SeratoMarkers2, but not in SeratoMarkers_,
-    // we could remove it from the output. We'll just leave it in for now.
+    // TODO(jholthuis): If a hotcue is set in SeratoMarkers2, but not in
+    // SeratoMarkers_, we could remove it from the output. We'll just leave it
```

Comment:
Why should we remove it?

Reply:
Because I'd think we should mirror Serato's behaviour when dealing with inconsistent data. The Markers tag has 5 fixed slots for hotcues and 3 fixed slots for loops, so if hotcue 3 is not set there's a specific byte pattern in the tag that tells us that it's not set and serato will ignore that hotcue. We should do the same.

Reply:
Okay, I did not get that this is to mirror Serato's behavior from the comment. Please include that information in the comment.